### DISCUSSION TITLE
Issue: DNS not working - Error: control/version.json | Couldn't get version check json ... /o timeout, couldn't initialize HTTP client or transport, cause: timeout exceeded: 18606 ms) | 502

### DISCUSSION THREAD

**Discussion by george298** [AdguardTeam/AdGuardHome#3038] [Q&A]
Hello,

Please help me fix this issue. I cannot resolve DNS after updating to v0.106.0

- [ v0.106.0] I am running the latest version
- [Yes ] I checked the documentation and found no answer
- [Yes] I checked to make sure that this issue has not already been filed

### Issue Details

Reported error message is below: 
Error: control/version.json | Couldn't get version check json from https://static.adguard.com/adguardhome/release/version.json: *fmt.wrapError updater: HTTP GET https://static.adguard.com/adguardhome/release/version.json: Get "https://static.adguard.com/adguardhome/release/version.json": synthetic.wrap: all upstreams failed to exchange request, cause: read udp [my ADH IP]:50072->9.9.9.9:53: i/o timeout (hidden: read udp [my ADH IP]:53514->149.112.112.112:53: i/o timeout, couldn't initialize HTTP client or transport, cause: timeout exceeded: 18606 ms) | 502

Traceroute output:
% traceroute 149.112.112.112
traceroute to 149.112.112.112 (149.112.112.112), 64 hops max, 52 byte packets
 1  [my router IP](my router IP)  1.646 ms  1.226 ms  0.543 ms
 2  * * *
 3  10.170.192.53 (10.170.192.53)  829.960 ms
    10.170.192.58 (10.170.192.58)  76.926 ms
    10.170.192.53 (10.170.192.53)  149.744 ms
 4  192.171.63.17 (192.171.63.17)  423.241 ms  222.732 ms
    192.171.61.161 (192.171.61.161)  33.156 ms
 5  206.248.189.97 (206.248.189.97)  32.890 ms  358.598 ms  583.135 ms

* **Version of AdGuard Home server:**
v0.106.0

* **How did you install AdGuard Home:**
Github releases script

* **How did you setup DNS configuration:**
Upstream DNS servers
9.9.9.9
149.112.112.112
https://dns.quad9.net/dns-query

Bootstrap DNS servers
9.9.9.10
149.112.112.10

* **If it's a router or IoT, please write device model:**
Raspberry Pi 3 Model B

* **CPU architecture:**
ARMV71

* **Operating system and version:**
Raspbian 10 (buster)

**Comment by ameshkov** [comment]
Try changing the upstream servers, Quad9 seems to not working from your place

**Comment by george298** [comment]
Thank you for your reply, I changed the DNS servers to Cloudfare and OpenDNS and performed a reboot of the raspberry pi for good measure -> this didn't work

I then stopped the service/uninstalled and downgraded back to v0.105.2 and restarted. Tried pressing Test Upstreams and got the same message as before:

Error: control/version.json | Couldn't get version check json from https://static.adguard.com/adguardhome/release/version.json: *fmt.wrapError updater: HTTP GET https://static.adguard.com/adguardhome/release/version.json: Get "https://static.adguard.com/adguardhome/release/version.json": synthetic.wrap: all upstreams failed to exchange request, cause: read udp [my ADH IP]:41467->1.1.1.1:53: i/o timeout (hidden: read udp [my ADH IP]41157->1.0.0.1:53: i/o timeout, read udp [my ADH IP]:40181->208.67.222.222:53: i/o timeout, read udp [my ADH IP]:33993->208.67.220.220:53: i/o timeout) | 502

Upstream DNS servers:
1.1.1.1
1.0.0.1
208.67.222.222
208.67.220.220

Bootstrap DNS servers
9.9.9.10
149.112.112.10
2620:fe::10
2620:fe::fe:10

I can successfully ping 1.1.1.1 but cannot resolve any DNS.

**Reply by ameshkov** [reply]
Weird, it looks like UDP is completely blocked for you.

What if you change your upstream to something TCP based? Like `https://1.1.1.1/dns-query`?

**Reply by george298** [reply]
Thank you! TCP-based works! Any idea why UDP would be blocked? I haven’t touched any rules on my firewall?.

**Reply by ameshkov** [reply]
I honestly have no clue, but it looks like exactly as it is blocked by a firewall (iptables or something like that).

**Comment by george298** [comment]
Server "1.0.0.1": could not be used, please check that you've written it correctly
Server "1.1.1.1": could not be used, please check that you've written it correctly
Server "208.67.220.222": could not be used, please check that you've written it correctly
Server "208.67.220.220": could not be used, please check that you've written it correctly
